src/main/java/org/mockito/internal/util/MockUtil.java

```diff
@@ -18,22 +18,55 @@
 import org.mockito.plugins.MockMaker.TypeMockability;
 import org.mockito.plugins.MockResolver;
 
+import java.util.Collections;
+import java.util.Map;
+import java.util.WeakHashMap;
 import java.util.function.Function;
 
 import static org.mockito.internal.handler.MockHandlerFactory.createMockHandler;
 
 @SuppressWarnings("unchecked")
 public class MockUtil {
 
-    private static final MockMaker mockMaker = Plugins.getMockMaker();
+    private static final MockMaker defaultMockMaker = Plugins.getMockMaker();
+    private static final Map<Class<? extends MockMaker>, MockMaker> mockMakers;
+
+    static {
+        mockMakers = Collections.synchronizedMap(new WeakHashMap<>());
+        mockMakers.put(defaultMockMaker.getClass(), defaultMockMaker);
+    }
 
     private MockUtil() {}
 
+    private static MockMaker getMockMaker(Class<? extends MockMaker> type) {
+        if (type == null) {
+            return defaultMockMaker;
+        } else {
```

Comment:
Nit: remove the else so that the return is on the same level as the if

Reply:
Done. For me, it feels easier to read with the else, but I guess that might just be because I am used to that style.